google/cloud/bigquery/table.py

```diff
@@ -636,12 +674,13 @@ def clustering_fields(self, value):
 
         (Defaults to :data:`None`).
         """
+        api_field = self._PROPERTY_TO_API_FIELD["clustering_fields"]
+
         if value is not None:
-            prop = self._properties.setdefault("clustering", {})
+            prop = self._properties.setdefault(api_field, {})
             prop["fields"] = value
         else:
-            if "clustering" in self._properties:
-                del self._properties["clustering"]
+            self._properties[api_field] = None
```

Comment:
Let's add a comment to explain why we need to unset the top-level field to `None`. (And thus can't use `_helpers._set_sub_prop` like some of the other properties in this class)